lib/phx_tools_web/plugs/curl_detector.ex

```diff
@@ -0,0 +1,49 @@
+defmodule PhxToolsWeb.CurlDetector do
+  @moduledoc """
+  A plug that serves a script based on the request's `User-Agent` or specific URL paths.
+
+  Requests from `curl` receive the script content directly, halting further processing. Additionally, requests to `/Linux.sh` or `/macOS.sh` paths return the script regardless of `User-Agent`.
+  """
+
+  import Plug.Conn
+
+  require Logger
+
+  @script_path "priv/script.sh"
+
+  @spec init(Keyword.t()) :: nil
+  def init(opts), do: nil
+
+  @spec call(Plug.Conn.t(), nil) :: Plug.Conn.t()
+  def call(conn, _opts) do
+    user_agent =
+      conn
+      |> get_req_header("user-agent")
+      |> List.first()
+
+    Logger.info("User-Agent: #{user_agent}==========")
+    path = conn.request_path
+    Logger.info("User-Agent: #{path}==========")
+
+    if should_serve_script?(user_agent, path) do
+      conn
+      |> put_resp_content_type("text/plain")
+      |> send_resp(200, script_content())
+      |> halt()
+    else
+      conn
+    end
+  end
+
+  defp should_serve_script?(user_agent, path) do
+    curl_user_agent?(user_agent) || path in ["/Linux.sh", "/macOS.sh"]
```

Comment:
Not sure if we need to cater for `/Linux.sh` and `/macOS.sh`. These are already files in our `priv/static` so the `Plug.Static` should take care of it:
https://github.com/optimumBA/phx.tools/blob/2ce95cbdf62c8ab0bae466664439a0f6278804b6/lib/phx_tools_web/endpoint.ex#L20-L24
https://github.com/optimumBA/phx.tools/blob/2ce95cbdf62c8ab0bae466664439a0f6278804b6/lib/phx_tools_web.ex#L21

Reply:
![Screenshot from 2024-11-14 18-03-44](https://github.com/user-attachments/assets/49ab3952-b860-4398-a945-6ff1676298c0)
This part in the requirements documents was a bit confusing

Reply:
In short, make sure all three of these work:
```bash
$SHELL -c "$(curl -fsSL https://phx-tools-pr-22.fly.dev)"
$SHELL -c "$(curl -fsSL https://phx-tools-pr-22.fly.dev/Linux.sh)"
$SHELL -c "$(curl -fsSL https://phx-tools-pr-22.fly.dev/macOS.sh)"
```

Reply:
Now it works.